pkg/util/cloudfoundry/garden_linux.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-Present Datadog, Inc.
+
+package cloudfoundry
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"code.cloudfoundry.org/garden"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// GetGardenContainers returns the list of running containers from the local garden API
+func (gu *GardenUtil) GetGardenContainers() ([]garden.Container, error) {
+	return gu.cli.Containers(nil)
+}
+
+// ListContainers returns the list of running containers and populates their metrics and metadata
+func (gu *GardenUtil) ListContainers() ([]*containers.Container, error) {
+	gardenContainers, err := gu.GetGardenContainers()
+	if err != nil {
+		return nil, fmt.Errorf("error listing garden containers: %v", err)
+	}
+
+	var cList = make([]*containers.Container, len(gardenContainers))
+	handles := make([]string, len(gardenContainers))
+	for i, gardenContainer := range gardenContainers {
+		handles[i] = gardenContainer.Handle()
+	}
+	gardenContainerInfo, err := gu.cli.BulkInfo(handles)
+	if err != nil {
+		return nil, fmt.Errorf("error getting info for garden containers: %v", err)
+	}
+	gardenContainerMetrics, err := gu.cli.BulkMetrics(handles)
+	if err != nil {
+		return nil, fmt.Errorf("error getting metrics for garden containers: %v", err)
+	}
+
+	for i, handle := range handles {
+		infoEntry := gardenContainerInfo[handle]
+		if err := infoEntry.Err; err != nil {
+			log.Debugf("could not get info for container %s: %v", handle, err)
+			continue
+		}
+		metricsEntry := gardenContainerMetrics[handle]
+		if err := metricsEntry.Err; err != nil {
+			log.Debugf("could not get metrics for container %s: %v", handle, err)
+			continue
+		}
+		container := containers.Container{
+			Type:        "garden",
+			ID:          handle,
+			EntityID:    containers.BuildTaggerEntityName(handle),
+			State:       infoEntry.Info.State,
+			Excluded:    false,
+			Created:     time.Now().Add(-metricsEntry.Metrics.Age).Unix(),
+			AddressList: parseContainerPorts(infoEntry.Info),
+		}
```

Comment:
I think the `Name` field is used to show the container in the live container view, not sure how this is going to be handled if we send it empty.
cc @DataDog/processes might know more about this

Reply:
They told me it's not used anymore, and that they get the name from the `container_name` tag

Reply:
didn't know that,  thanks 👍